Model/Service/DumpCmsDataService.php

```diff
@@ -76,6 +78,34 @@ private function write(WriteInterface $writeDirectory, string $filePath, string
         $stream->close();
     }
 
+    private function getBlocksMapping(): array
+    {
+        $blocksMapping = [];
+        $searchCriteria = $this->criteriaBuilder;
+        $blocksList = $this->blockRepository->getList($searchCriteria->create());
```

Comment:
You are pulling every block no matter how many are we actually exporting. Refactor this:
- Move calling this outside of __construct()
- add pagination support when pulling the data
- Call this method inside of replaceBlockIds() (pull all IDs for every call of replaceBlockId()), use the ->blocksMapping as a cache so it doesn't pull the same blockIDs over and over again